### ISSUE TITLE
Long sentence list titles overflow header container

### ISSUE THREAD

**Issue by cblanken** [Tatoeba/tatoeba2#3118]
**Long sentence list titles overflow header container**

## Examples
See [CK's List 907](https://tatoeba.org/en/sentences_lists/show/907)
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/c00768e2-ccee-4fb0-bf7e-be15d04df7de)

## Another Example
It's even more pronounced in this [list](https://tatoeba.org/en/sentences_lists/show/3136)
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/2d6cc03e-ca02-470b-bd79-f6567e026aa1)

**Comment by jiru** [comment]
> Thoughts?

I think we should add a "list description" kind of field to allow users to describe the list using as much text as they want, and restrict the number of characters of the list title to a reasonable value, possibly truncating it using CSS ellipsis or something. A title that is too long loose its purpose of summarizing the list purpose.

**Comment by cblanken** [comment]
> ...possibly truncating it using CSS ellipsis or something. A title that is too long loose its purpose of summarizing the list purpose.

I agree, it'd definitely be better to have a separate field for longer description. If the title is truncated for existing lists though, it may degrade the user experience a bit, since some lists (I'm mainly looking at CK's lists here) have similar prefixes which may make it difficult to identify which list is currently open depending on where the cutoff/ellipsis happens. We could add a popup on hover to display the whole title, but I don't think that would be ideal.

## Some examples of truncated titles
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/6912c748-e229-4f2d-9b69-ddc80041d1e2)

![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/422f514e-748c-4c65-b169-bbc166eb0f85)

![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/99e0417b-87b6-48f0-8970-474602195074)

![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/8b58b555-9c33-4805-b7ee-8da78ac20ffc)

**Comment by jiru** [comment]
Yeah maybe the ellipsis is not the best way to go about it, especially giving that the available space varies depending on how "sort by" is translated.

We may as well simply ask individual contributors to update their list names. I just checked the database, as for lists titled with more than 100 chars, CK owns most of them, and then there are lists 992, 170438, 169999 and 170736.

**Comment by jiru** [comment]
> Wrapping once and then truncating with an ellipsis actually seems to be a bit complicated.
> [...]
> Here's what they look like with anything over two line hidden.

I think truncating two lines is still a big improvement over the messy overflow we currently have. If you want to add the CSS code for ellipsis, I suggest using the trick provided by vinesh on the StackOverflow page. Even though `-webkit-box-orient` is not great, it’s currently widely supported (even by Firefox), and even if it stops working in the future, it’s not a big deal that the ellipsis simply disappear.

Also please make sure the title wraps okay in languages that do not use space as word delimiter, such as Chinese or Japanese.

**Comment by cblanken** [comment]
Something to keep in mind. We lose a lot of space (down to about 50 characters) when the list is owned by the user due to the extra controls they have. See examples below.

## List page for list not owned by logged in user
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/ca3d9858-4517-4879-ae64-27c7570fcd37)

## List page for list owned by logged in user
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/7d4d4140-d891-499a-8755-1bda1762e307)

**Comment by cblanken** [comment]
> I suggest using the trick provided by vinesh on the StackOverflow page. Even though `-webkit-box-orient` is not great, it’s currently widely supported (even by Firefox), and even if it stops working in the future, it’s not a big deal that the ellipsis simply disappear.

One setback I noticed with this solution is that any titles that don't require wrapping will offset to the top as if expecting a 2nd line instead of remaining centered.

## Before
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/5d156f97-0b5e-4aec-83c0-8e657a28bcea)

## After
![image](https://github.com/Tatoeba/tatoeba2/assets/19908880/f771d2c0-2541-47b7-8b93-a468dae538b2)

I don't think it's possible to get that re-centered with just CSS. It would probably require some Javascript to check the title content and somehow measure the text width before applying the wrapping/truncation styles.
